### PR TITLE
Add call home to Artifactory

### LINKED ISSUE #90: Add User-Agent string to Artifactory API HTTP request header
UA should be in the form of `jfrog/vault-plugin-secrets-artifactory/0.2.6`

### PR DESCRIPTION
**Pull request by alexhung**
Call home request does not contain any data from the users. It sends only the product ID (e.g. `vault-plugin-secrets-artifactory/0.2.18`) and the Vault actions (e.g. `pathConfigRotateUpdate`) to Artifactory call home API endpoint.

Also fixes #90

### CODE REVIEW

**Review by MahithaB — DISMISSED** [review]

**Comment by TJM** [comment]
I would strongly recommend making this able to be opt-out, many secure environments won't have the ability to reach random sites on the Internet. Additionally, some people may have privacy concerns with any sort of "callHome" regardless of  what data is included.

File: artifactory.go [hunk 1]
```diff
@@ -347,6 +346,44 @@ func (b *backend) getRootCert(config adminConfiguration) (cert *x509.Certificate
...
+	FeatureId string `json:"featureId"`
+}
+
+type Usage struct {
+	ProductId string    `json:"productId"`
+	Features  []Feature `json:"features"`
+}
+
+func (b *backend) callHome(config adminConfiguration, featureId string) {
+	b.Backend.Logger().Debug("callHome")
```
Comment (TJM) [inline artifactory.go:359]: This should probably have the featureId string?
```suggestion
	b.Backend.Logger().Debug("callHome", "featureId", featureId)
```

File: artifactory.go [hunk 2]
```diff
@@ -347,6 +346,44 @@ func (b *backend) getRootCert(config adminConfiguration) (cert *x509.Certificate
...
+	if err != nil {
+		b.Backend.Logger().Info("error marshalling call home request", "err", err)
+		return
+	}
+
+	resp, err := b.performArtifactoryPostWithJSON(config, "artifactory/api/system/usage", jsonReq)
+	if err != nil {
+		b.Backend.Logger().Info("error making call home request", "response", resp, "err", err)
+		return
+	}
```
Comment (TJM) [inline artifactory.go:381]: Well, now I am confused. This appears to be posting this JSON data to the artifactory server itself. Although that API doesn't appear in the [API docs](https://www.jfrog.com/confluence/display/rtf/artifactory+rest+api). I assume there is some sort of "call home relay" ?

File: artifactory.go (hunk 1 above)
Comment (alexhung) [reply artifactory.go:359]: Actually I meant to remove this before opening the PR.

**Review by danielmkn — APPROVED** [review]

**Pull request merged**